### ISSUE TITLE
Building servers in cpprestsdk

### ISSUE THREAD

**Issue by Maktm** [Microsoft/cpprestsdk#511]
Is building servers (as opposed to HTTP clients) considered a risky thing to do with the use of cpprestsdk? I'm asking because I checkout the [documentation page](https://casablanca.codeplex.com/documentation) (which may be un-updated) and it says that the HTTP listener is still in beta stage.

**Comment by LeonineKing1199** [comment]
It just has really poor perf, I think :P

**Comment by pcaluke** [comment]
We've created a micro service server application which runs on windows server with reasonable performance and stability. We get a 2ms response time from the listener but it seems to top out at 575 threads for some reason. Running it on Ubuntu we get around a 50ms response time

**Comment by LeonineKing1199** [comment]
Yeah, the suspect in that case might actually be the threadpool implementation that MS has for Linux. On Windows, it hooks into the system threadpool which I'm assuming is more optimized than whatever cpprestsdk has.

This is unfortunate for several reasons. The number one being, you're most likely going to be running your server not on Windows but rather Linux.

**Comment by fcharlie** [comment]
I use cpprestsdk write a `Git LFS Server`, but CPU usage is hight, cpprestsdk use Boost.Asio on Linux, but missing `sendfile`。
pplx performance is not very good in linux

**Comment by sanjeev-at-runnertech** [comment]
Will this mean that we cannot use it for mainstream applications and go to production?

**Comment by LeonineKing1199** [comment]
Pretty much. Rust and Go have better and more machine-efficient back-end frameworks than anything else in C++ right now.

C++ has Asio which is fast but can be quite an uphill battle to learn and master.
